### DISCUSSION TITLE
Telegram Mini App: Wallet Loader Not Displaying, Causing Redirection Issues on First Load. In demo app there is loaders beside wallets, how i can achieve this ?

### DISCUSSION THREAD

**Discussion by mitulmlakhani** [reown-com/appkit#3354] [Q&A]
I'm building the Telegram mini app with AppKit and following the React example with Ethers.

When the wallet modal opens, none of the wallets appear ready. In the demo Telegram mini app (available [here](https://t.me/appkit_test_ggr_bot)), a loader is displayed next to the wallet list for 3-5 seconds while the wallets are loading.

![Screenshot 2024-11-29 at 2 26 44 AM](https://github.com/user-attachments/assets/ab31a0f0-bef5-4947-ad45-4c0e28e1590b)

However, in my app, no loaders are visible. If users click a wallet before it's ready, the redirection fails. Clicking after 5 seconds works fine.
Note: This issue occurs only on the first load.

Note: this issue getting first time only and on Iphone only

Packages I'm Using:

- @reown/appkit: "^1.5.0"
- @reown/appkit-adapter-ethers: "^1.5.0"
- @walletconnect/modal-core: "^2.7.0"
- @walletconnect/modal-ui: "^2.7.0"
- ethers: "^6.13.4"

**- react app AppKit intialize**       

```
  createAppKit(newConfig);
       
  const newConfig: any = {
    adapters: [ethersAdapter],
    networks,
    metadata,
    projectId,
    features: {
      analytics: true,
      email: false,
      onramp: false,
      socials: [],
    },
    // defaultNetwork: mainnetChain,
  };
```
  
AppKit Component:
`<appkit-button />  `

**Comment by arein** [comment]
thanks for reporting - what app are you building? we'd love to see it

@ganchoradkov can you pls advise?

**Reply by mitulmlakhani** [reply]
Just doing wallet connection for evm using ethersjs

**Reply by arein** [reply]
yes - we want to see the app :)

**Reply by mitulmlakhani** [reply]
https://t.me/CrappyGameBot/Play

- following this docs with react: https://docs.reown.com/appkit/features/telegram-mini-apps

**Reply by mitulmlakhani** [reply]
you can see web version from here: https://sweet-swan-46f670.netlify.app/

**Reply by arein** [reply]
cool - curious: why do you not have social login enabled?

**Reply by mitulmlakhani** [reply]
They are not required on the project, FYI on enabling getting the same behavior.

**Comment by mitulmlakhani** [comment]
code is the same as https://github.com/reown-com/appkit-web-examples/tree/main/react/react-ethers from @rtomas  code samples.

**Comment by arein** [comment]
@mitulmlakhani we haven't released this yet - https://github.com/reown-com/appkit/pull/3025 - coming shortly

**Reply by mitulmlakhani** [reply]
I'm facing issue on wallet(metamask) redirect.

**Reply by mitulmlakhani** [reply]
Hi @arein , can you share code repo of this https://t.me/appkit_test_ggr_bot. I can't find it.

**Reply by rtomas** [reply]
there is no special code ... please check our guide for mini apps -> https://docs.reown.com/appkit/recipes/telegram-mini-app
